### DISCUSSION TITLE
Change graph's axis font

### DISCUSSION THREAD

**Discussion by RubenAntunes16** [patrykandpatrick/vico#641] [Questions] [answered]
Hello! 👋
Is it possible to change the graph's axis font? If so, how can I do it?

Thanks! 🙂

**Comment by patrickmichalik** [comment]
Hello! Yup, this is possible. Are you using Vico 1 or Vico 2, and are you using it with Jetpack Compose or with the view system?

**Reply by RubenAntunes16** [reply]
I'm using Vico 2 with Jetpack compose

**Reply by patrickmichalik** [reply]
Thanks, @RubenAntunes16! It looks like @kevincianfarini has already shared a solution for this setup below.

**Comment by kevincianfarini** [comment, marked as answer]
If you're using Compose, we convert a `TextStyle` to a `Typeface` by doing the following: 

```kt
@Composable
internal fun TextStyle.toGraphicsTypeFace(): Typeface {
    val resolver = LocalFontFamilyResolver.current
    return remember(resolver, this) {
        resolver.resolve(
            fontFamily = this.fontFamily,
            fontWeight = this.fontWeight ?: FontWeight.Normal,
            fontStyle = this.fontStyle ?: FontStyle.Normal,
            fontSynthesis = this.fontSynthesis ?: FontSynthesis.All,
        )
    }.value as Typeface
}
```

and we then set the fontface by using: 

```kt
startAxis = rememberStartAxis(
    label = rememberTextComponent(
        color = axisLabelTextColor,
        textSize = axisLabelTextStyle.fontSize,
        typeface = axisLabelTextStyle.toGraphicsTypeFace(),
    ),
)
```

**Reply by kevincianfarini** [reply]
Perhaps it makes sense to migrate the Vico Compose functions taking a `Typeface` to use `TextStyle` using this approach 😄

**Reply by RubenAntunes16** [reply]
Thanks a lot for your suggestion! 🙌🏻

**Reply by kevincianfarini** [reply]
No problem. @patrickmichalik and/or @Gowsky, would you accept a PR for an overload of `rememberTextComponent` that accepts a compose `TextStyle` instead of `android.graphics.Typeface`? eg. 

```kt
@Composable
public fun rememberTextComponent(
    color: Color = Color.Black,
    background: ShapeComponent? = null,
    ellipsize: TextUtils.TruncateAt = TextUtils.TruncateAt.END,
    lineCount: Int = Defaults.LABEL_LINE_COUNT,
    padding: MutableDimensions = emptyDimensions(),
    margins: MutableDimensions = emptyDimensions(),
    textStyle: TextStyle? = null,
    textAlignment: Layout.Alignment = Layout.Alignment.ALIGN_NORMAL,
    minWidth: TextComponent.MinWidth = TextComponent.MinWidth.fixed(),
): TextComponent = rememberTextComponent(
    color = color,
    textSize = textStyle.fontSize ?: Defaults.TEXT_COMPONENT_TEXT_SIZE.sp,
    background = background,
    ellipsize = ellipsize,
    lineCount = lineCount,
    padding = padding,
    margins margins,
    typeface = textStyle?.toGraphicsTypeface(),
    textAlignment = textAlignment,
    minWidth = minWidth,
)
```

**Reply by patrickmichalik** [reply]
Certainly, @kevincianfarini! Thanks for the help and for your interest in contributing. It would be optimal to add such an overload for `rememberAxisLabelComponent` too. You can do so if you’re interested, or we can take care of it after merging your PR.

**Reply by kevincianfarini** [reply]
https://github.com/patrykandpatrick/vico/pull/650

**Reply by patrickmichalik** [reply]
Cheers, @kevincianfarini! I’ll take a look shortly.
